optimum/habana/transformers/generation/utils.py

```diff
@@ -664,6 +629,7 @@ def generate(
                 decoder_start_token_id=generation_config.decoder_start_token_id,
                 bos_token_id=generation_config.bos_token_id,
                 device=inputs_tensor.device,
+                max_new_tokens=generation_config.max_new_tokens,
```

Comment:
Related to my comment above, we can set `self.generation_config.max_new_tokens = generation_config.max_new_tokens` to access it inside the method. WDYT?